src/Authentik/Provider.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace SocialiteProviders\Authentik;
+
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'AUTHENTIK';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = ['openid goauthentik.io/api profile email'];
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function additionalConfigKeys()
+    {
+        return ['base_url'];
+    }
+
+    protected function getBaseUrl()
+    {
+        return rtrim($this->getConfig('base_url'), '/');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase($this->getBaseUrl().'/application/o/authorize/', $state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return $this->getBaseUrl().'/application/o/token/';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get($this->getBaseUrl().'/application/o/userinfo/', [
+            'headers' => [
```

Comment:
Please use the RequestOptions headers const instead

Reply:
I don't know what you mean by this one - are you able to give an example?

Reply:
Check other providers

Reply:
Done